handlers/authentication.go

```diff
@@ -249,6 +249,39 @@ func performOpenIdAuthentication(w http.ResponseWriter, r *http.Request) bool {
 			RespondWithDetailedError(w, http.StatusForbidden, "the OpenID token was rejected", err.Error())
 			return true
 		}
+
+		if len(conf.Auth.OpenId.AllowedDomains) > 0 {
+			log.Debugf("Allowed Domains: %v", conf.Auth.OpenId.AllowedDomains)
+			tokenClaims := openIdParams.ParsedIdToken.Claims.(jwt.MapClaims)
+			var hostedDomain string
+			if v, ok := tokenClaims["hd"]; ok {
+				hostedDomain = v.(string)
+			} else {
+				splitedSubject := strings.Split(openIdParams.Subject, "@")
```

Comment:
@vasartori I wonder if you should use `tokenClaims["email"]` instead of `openIdParams.Subject`. 

Reading this Google docs: https://developers.google.com/identity/protocols/oauth2/openid-connect
I see the "sub" (Subject) is an opaque ID. So, it's probably better to use the `email` claim --- although you will need to check for it's presence.

Reply:
@israel-hdez you are right, my mistake.
I'll change the code

Reply:
@israel-hdez changed made.